### DISCUSSION TITLE
New MCU showing "offline" while sleeping when on 4G

### DISCUSSION THREAD

**Discussion by espenu** [teslamate-org/teslamate#1239] [Q&A]
**Describe the bug**

After upgrading my car to MCU2, the sleep state is shown as "offline" when I'm not on WiFi.
When on WiFi it works fine and is correctly shown as "asleep".
Note that when I had MCU1 my car never fell asleep, so this issue may have been present before without me noticing.

**Expected behavior**

Should show as asleep regardless of being on WiFi or 4G

**How to reproduce it (as minimally and precisely as possible)**:

For me, reproducing is as simple as leaving the car somewhere without WiFi for a couple of hours.

**Relevant entries from the logs**
Note that it did not start showing offline in states until 18:27:32

```
teslamate_1           | 2021-01-13 18:06:48.705 car_id=1 [info] Start / :online
teslamate_1           | 2021-01-13 18:06:48.724 car_id=1 [info] Connecting ...
teslamate_1           | 2021-01-13 18:09:54.658 car_id=1 [info] Suspending logging
teslamate_1           | 2021-01-13 18:11:32.326 [info] Connection was closed (a:1|t:1|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:11:43.221 [info] Connection was closed (a:1|t:2|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:16:31.374 [info] Connection was closed (a:1|t:1|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:16:42.283 [info] Connection was closed (a:1|t:2|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:21:31.266 [info] Connection was closed (a:1|t:1|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:21:42.231 [info] Connection was closed (a:1|t:2|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:21:53.125 [info] Connection was closed (a:1|t:3|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:22:04.014 [info] Connection was closed (a:1|t:4|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:22:20.731 car_id=1 [info] Fetching vehicle state ...
teslamate_1           | 2021-01-13 18:22:20.908 [info] Connection was closed (a:1|t:5|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:22:21.629 [info] control:hello – 0
teslamate_1           | 2021-01-13 18:22:51.630 [info] Stream.Timeout / 5
teslamate_1           | 2021-01-13 18:22:51.809 [info] Connection was closed (a:1|t:6|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:22:52.789 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:23:22.790 [info] Stream.Timeout / 6
teslamate_1           | 2021-01-13 18:23:22.969 [info] Connection was closed (a:1|t:7|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:23:23.964 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:23:53.965 [info] Stream.Timeout / 7
teslamate_1           | 2021-01-13 18:23:54.141 [info] Connection was closed (a:1|t:8|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:23:55.114 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:24:25.115 [info] Stream.Timeout / 8
teslamate_1           | 2021-01-13 18:24:25.293 [info] Connection was closed (a:1|t:9|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:24:26.257 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:24:56.258 [info] Stream.Timeout / 9
teslamate_1           | 2021-01-13 18:24:56.435 [info] Connection was closed (a:1|t:10|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:24:57.381 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:25:27.382 [info] Stream.Timeout / 10
teslamate_1           | 2021-01-13 18:25:27.559 [info] Connection was closed (a:1|t:11|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:25:28.486 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:25:58.487 [info] Stream.Timeout / 11
teslamate_1           | 2021-01-13 18:25:58.664 [info] Connection was closed (a:1|t:12|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:25:59.695 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:26:29.696 [info] Stream.Timeout / 12
teslamate_1           | 2021-01-13 18:26:29.873 [info] Connection was closed (a:1|t:13|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:26:30.809 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:27:00.809 [info] Stream.Timeout / 13
teslamate_1           | 2021-01-13 18:27:00.987 [info] Connection was closed (a:1|t:14|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:27:02.010 [error] Vehicle Error: Vehicle is offline
teslamate_1           | 2021-01-13 18:27:32.010 [info] Stream.Timeout / 14
teslamate_1           | 2021-01-13 18:27:32.010 car_id=1 [info] Fetching vehicle state ...
teslamate_1           | 2021-01-13 18:27:32.188 [info] Connection was closed (a:1|t:15|d:1). Reconnecting …
teslamate_1           | 2021-01-13 18:27:32.795 car_id=1 [info] Start / :offline
teslamate_1           | 2021-01-13 18:27:32.803 car_id=1 [info] Disconnecting ...
```

**Screenshots**
Screenshot showing part of my day today. sleeping while at home on WiFi, then going offline with the same intervals as when sleeping when I'm at work.

![image](https://user-images.githubusercontent.com/1891108/104487200-142bbf00-55cd-11eb-9970-9da0b88fdd91.png)

**Data**
States table from database for today:

1325,online,2021-01-13 03:00:30.809,2021-01-13 03:16:32.486323,1
1326,asleep,2021-01-13 03:16:32.486323,2021-01-13 04:15:38.369,1
1327,online,2021-01-13 04:15:38.369,2021-01-13 04:31:33.45213,1
1337,online,2021-01-13 11:13:11.793,2021-01-13 11:29:20.110467,1
1340,offline,2021-01-13 12:49:40.016239,2021-01-13 13:43:45.423,1
1342,offline,2021-01-13 15:08:15.805682,2021-01-13 15:51:45.856,1
1343,online,2021-01-13 15:51:45.856,2021-01-13 16:12:31.725538,1
1322,asleep,2021-01-13 00:46:10.716873,2021-01-13 01:45:18.443,1
1323,online,2021-01-13 01:45:18.443,2021-01-13 02:01:23.453075,1
1338,offline,2021-01-13 11:29:20.110467,2021-01-13 12:28:28.131,1
1345,online,2021-01-13 17:06:48.298,2021-01-13 17:27:32.795652,1
1332,offline,2021-01-13 07:44:02.116716,2021-01-13 08:43:08.958,1
1333,online,2021-01-13 08:43:08.958,2021-01-13 09:04:32.771329,1
1334,offline,2021-01-13 09:04:32.771329,2021-01-13 09:58:06.409,1
1341,online,2021-01-13 13:43:45.423,2021-01-13 15:08:15.805682,1
1346,offline,2021-01-13 17:27:32.795652,,1
1329,online,2021-01-13 05:30:39.126,2021-01-13 06:33:55.907706,1
1331,online,2021-01-13 07:28:00.659,2021-01-13 07:44:02.116716,1
1335,online,2021-01-13 09:58:06.409,2021-01-13 10:19:06.890721,1
1336,offline,2021-01-13 10:19:06.890721,2021-01-13 11:13:11.793,1
1324,asleep,2021-01-13 02:01:23.453075,2021-01-13 03:00:30.809,1
1328,asleep,2021-01-13 04:31:33.45213,2021-01-13 05:30:39.126,1
1330,offline,2021-01-13 06:33:55.907706,2021-01-13 07:28:00.659,1
1339,online,2021-01-13 12:28:28.131,2021-01-13 12:49:40.016239,1
1344,offline,2021-01-13 16:12:31.725538,2021-01-13 17:06:48.298,1

**Environment**

- TeslaMate version: 1.21.1
- Type of installation: Docker (with database server manualy installed)
- OS TeslaMate is installed on: Ubuntu container
- User OS & Browser: Windows10 & Manjaro + Chrome & Vivaldi

**Comment by cwanja** [comment]
What Tesla software version are you on? I saw the same behavior randomly and then after an update it went back to ‘asleep’.

**Comment by w3zly** [comment]
I got the same too - MCU2 upgrade with latest update 2020.48.30. I turned of Wifi to try and identify why my car sleeps for an hour and then wakes for 15 mins all the time. In the time that i was using LTE on, sleeping was reported as offline
![image](https://user-images.githubusercontent.com/75405351/104853576-b1526480-58f9-11eb-93d4-997f0e0d47cc.png)

**Comment by espenu** [comment]
> What Tesla software version are you on? I saw the same behavior randomly and then after an update it went back to ‘asleep’.

While on MCU 2 I've had 44.25, 48.10, 48.12.1, 48.26, and now 48.30. All of them have had the same issue.

> I got the same too - MCU2 upgrade with latest update 2020.48.30. I turned of Wifi to try and identify why my car sleeps for an hour and then wakes for 15 mins all the time. In the time that i was using LTE on, sleeping was reported as offline

I think the 15 min awake is to contact the mothership to check for updates and stuff. After a while it should go into a deeper sleep (at least that's what I've heard).

**Comment by cwanja** [comment]
> While on MCU 2 I've had 44.25, 48.10, 48.12.1, 48.26, and now 48.30. All of them have had the same issue.
> 
I never got an upgrade, just had MCU 2 since I got the car. There _might_ be something with that upgrade, but I know it is not TeslaMate related. It eventually goes back to "sleeping" as opposed to "offline"
>I think the 15 min awake is to contact the mothership to check for updates and stuff. After a while it should go into a deeper sleep (at least that's what I've heard).
>
It does, after about 7 - 10, 15 - 20 minutes each cycle it will fall into a deep sleep. Used to be for almost 48 hours, now I am seeing it at 24 hours.

**Comment by w3zly** [comment]
> > While on MCU 2 I've had 44.25, 48.10, 48.12.1, 48.26, and now 48.30. All of them have had the same issue.
> 
> I never got an upgrade, just had MCU 2 since I got the car. There _might_ be something with that upgrade, but I know it is not TeslaMate related. It eventually goes back to "sleeping" as opposed to "offline"
> 
> > I think the 15 min awake is to contact the mothership to check for updates and stuff. After a while it should go into a deeper sleep (at least that's what I've heard).
> 
> It does, after about 7 - 10, 15 - 20 minutes each cycle it will fall into a deep sleep. Used to be for almost 48 hours, now I am seeing it at 24 hours.

Hmm...mine has always done the sleep for an hour, wake for 15 mins...never has it slept for more then an hour before waking even if the car is sat on the drive for a couple days and not driven or checked with the official Tesla app.

Not sure if this could be done, but it would be useful if we could identify via Teslamate when it sleeps normally Vs deep sleep...I believe teslafi does this?

**Comment by cwanja** [comment]
> Not sure if this could be done, but it would be useful if we could identify via Teslamate when it sleeps normally Vs deep sleep...I believe teslafi does this?
>
Really? You have never had it go into a long stretch of sleep? Similar to this (this is a portion of a seven day graphic). Also pretty sure that the end of it, it is in a "deep sleep":
![Screen Shot 2021-01-18 at 8 45 47 AM](https://user-images.githubusercontent.com/54645561/104929483-91f32000-5969-11eb-8371-1ee4c065e57e.png)

Not sure if there is something in the API that shows deep versus normal sleeping (@adriankumpf), but I know TeslaMate's dashboard seems to show varying amounts of data when it sleeps and when it shows the least amount of data, I know it will keep sleeping for hours.

**Comment by w3zly** [comment]
> > Not sure if this could be done, but it would be useful if we could identify via Teslamate when it sleeps normally Vs deep sleep...I believe teslafi does this?
> 
> Really? You have never had it go into a long stretch of sleep? Similar to this (this is a portion of a seven day graphic). Also pretty sure that the end of it, it is in a "deep sleep":
> ![Screen Shot 2021-01-18 at 8 45 47 AM](https://user-images.githubusercontent.com/54645561/104929483-91f32000-5969-11eb-8371-1ee4c065e57e.png)
> 
> Not sure if there is something in the API that shows deep versus normal sleeping (@adriankumpf), but I know TeslaMate's dashboard seems to show varying amounts of data when it sleeps and when it shows the least amount of data, I know it will keep sleeping for hours.

Nope :( never...This is 7 days...

![image](https://user-images.githubusercontent.com/75405351/104937563-d2708f00-59a5-11eb-83a3-ecea721f9add.png)

**Comment by espenu** [comment]
> Nope :( never...This is 7 days...

You're doing daily drives though. I would try and leave it for more than a day without driving if possible to see if it goes into deep sleep then. Also, not sure if being connected to power has an effect since it would not be so critical to deep sleep when power is available. But this sleeping thing is new to me too, so I'm no expert.

**Comment by w3zly** [comment]
> > Nope :( never...This is 7 days...
> 
> You're doing daily drives though. I would try and leave it for more than a day without driving if possible to see if it goes into deep sleep then. Also, not sure if being connected to power has an effect since it would not be so critical to deep sleep when power is available. But this sleeping thing is new to me too, so I'm no expert.

I've done a few days with no driving and not plugged in before - still does the same 1 hour on then 15 mins awake - have changed tokens also to ensure no 3rd party apps are requesting access...still no effect...i guess ill have to keep experimenting! (or wait for the next vehicle update and hope it resolves it!)

**Comment by cwanja** [comment]
Going to ask the awful questions - have you tried a reboot? Have you also tried to power off the vehicle for three - five minutes? 

You could ask Tesla to investigate, but the are going to adamantly say it’s the third-party app. 

I will also be completely honest - never have I attempted to let my Tesla sleep on cellular because if it’s not home, Sentry Mode is on. Does it show offline at home if you kill WiFi on the Tesla?

**Comment by 65sc02** [comment]
Hello,

During the last 7 days my MS75D was mostly asleep (good) but about every 2 days it wakes up and stays awake for about 12 hours:



Is this normal behavior? I notice a drop in the SoC (1% to 2% or so) every time it stays awake for such long periods. I would like to avoid this.

I would prefer it if the car would just keep sleeping or only wake up for 15 minutes and then go back to sleep. Why does it stay awake for 12 hours??

hh

**Comment by cwanja** [comment]
> Hello, During the last 7 days my MS75D was mostly asleep (good) but about every 2 days it wakes up and stays awake for about 12 hours: Is this normal behavior? I notice a drop in the SoC (1% to 2% or so) every time it stays awake for such long periods. I would like to avoid this. I would prefer it if the car would just keep sleeping or only wake up for 15 minutes and then go back to sleep. Why does it stay awake for 12 hours?? hh

I think what you are seeing is normal software update behavior. Seen my 2019 Model S randomly stay awake for 12 hours after some updates and then sleep peacefully for days with some updates. It is then my belief that something in the software update throws off the coding for the sleep pattern and thus the inconsistent sleeping. 

Maybe bias, but I would say that this seemingly effects S and X owners more than 3 and Y owners. Just by following comments here and on Reddit.

**Comment by espenu** [comment]
@adriankumpf Why was this issue closed? I see that the discussion veered off to something unrelated, but isn't the original issue a valid one?
By the way, I've seen that my car is now often starting to show offline even when it's on WiFi (or at least should be. I can't really check because then I will have woken it up and invalidated the test).

**Reply by adriankumpf** [reply]
The issue wasn't closed. I moved it to "Discussions" as this is not a TeslaMate bug. If the Tesla API reports your car as "offline", TeslaMate will display it that way.

**Comment by fetzu** [comment]
I'll allow myself to “revive” this discussion and share what has been happening with my car ever since I upgraded to MCU2 (done in conjunction with the AP3 upgrade) last may.

I have documented my trials (and tribulations) on [TMC](https://teslamotorsclub.com/tmc/posts/5971077/). That post and the follow-ups have details regarding the situation, but to put it short:

- **My car will not sleep, it just goes through long strides of alternating online/offline states**.
- I used the bug (dynotest) a few FW versions ago to confirm that **my car has never slept more than 3941 seconds since it first started back in May**. The same menu shown that my car slept a total of 2838940 seconds (that's a little over 32 days in just shy of a 4 months period). Please note that these stats were collected before I changed my API keys/turned off all logging, and that I have no access to that menu anymore.
- **I have been experiencing abnormal vampire drain/phantom loss since that MCU2 upgrade**; I was experiencing an average of about 0.2 km/h on MCU1, I am at something around 0.7 km/h on MCU2.
- As I documented in my posts, **I tried all the usual suspects** (from checking the sentry/summon standby/USB devices, to all kinds of reboots, to invalidating all my API keys multiple times through password changes).

I recently experimented with different “logging” conditions, and concluded that:
- **Turning off ALL logging decreases my vampire drain/phantom loss tremendously** (to around 0.17 km/h).
- **Using TeslaMate only yields a little more vampire drain/phantom loss** (around 0.35 km/h)
- **Using TeslaFi gives me that crazy drain of 0.7 km/h on average**.

Please note that **turning all logging off causes my car's LTE to disconnect in some cases** (i.e., be unavailable when I get in the car and taking around 5 minutes to re-connect), **this never happen with a logging app turned on**.

**I have been in contact with Tesla support**, but their advice is “don't use any third-party services” (because of course it is), and they will be looking into the LTE disconnecting issue “as soon as there is some spare time”. So I am not holding my breath for help from their side.

Here is what the "idle" patterns looks like (with only TeslaMate turned on):
![image](https://user-images.githubusercontent.com/6372605/137014177-16bbb8ce-b165-4032-b3ae-29bada39b74e.png)

And here is a sample of how my TM logs look like:
<html><body>


date | steam | message
-- | -- | --
2021-10-11 14:02:56 | stdout | 2021-10-11 16:02:56.639 car_id=1 [info] Start / :online
2021-10-11 14:02:57 | stdout | 2021-10-11 16:02:57.058 car_id=1 [info] Connecting ...
2021-10-11 14:06:00 | stdout | 2021-10-11 16:06:00.947 car_id=1 [info] Suspending logging
2021-10-11 14:17:21 | stdout | 2021-10-11 16:17:21.425 car_id=1 [info] Fetching vehicle state ...
2021-10-11 14:17:21 | stdout | 2021-10-11 16:17:21.851 [error] Vehicle Error: Vehicle is offline
2021-10-11 14:17:21 | stdout | 2021-10-11 16:17:21.853 car_id=1 [info] Start / :offline
2021-10-11 14:17:21 | stdout | 2021-10-11 16:17:21.947 car_id=1 [info] Disconnecting ...
2021-10-11 15:16:24 | stdout | 2021-10-11 17:16:24.032 car_id=1 [info] Start / :online
2021-10-11 15:16:24 | stdout | 2021-10-11 17:16:24.710 car_id=1 [info] Connecting ...
2021-10-11 15:19:28 | stdout | 2021-10-11 17:19:28.124 car_id=1 [info] Suspending logging
2021-10-11 15:29:58 | stdout | 2021-10-11 17:29:58.007 car_id=1 [info] Fetching vehicle state ...
2021-10-11 15:30:59 | stdout | 2021-10-11 17:30:59.416 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:31:29 | stdout | 2021-10-11 17:31:29.767 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:32:00 | stdout | 2021-10-11 17:32:00.289 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:32:30 | stdout | 2021-10-11 17:32:30.742 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:33:01 | stdout | 2021-10-11 17:33:01.119 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:33:31 | stdout | 2021-10-11 17:33:31.683 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:34:04 | stdout | 2021-10-11 17:34:04.649 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:34:35 | stdout | 2021-10-11 17:34:35.251 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:35:05 | stdout | 2021-10-11 17:35:05.252 car_id=1 [info] Fetching vehicle state ...
2021-10-11 15:35:05 | stdout | 2021-10-11 17:35:05.717 car_id=1 [info] Start / :offline
2021-10-11 15:35:05 | stdout | 2021-10-11 17:35:05.874 [error] Vehicle Error: Vehicle is offline
2021-10-11 15:35:05 | stdout | 2021-10-11 17:35:05.909 car_id=1 [info] Disconnecting ...
2021-10-11 16:29:58 | stdout | 2021-10-11 18:29:58.433 car_id=1 [info] Start / :online
2021-10-11 16:29:58 | stdout | 2021-10-11 18:29:58.649 car_id=1 [info] Connecting ...
2021-10-11 16:33:03 | stdout | 2021-10-11 18:33:03.563 car_id=1 [info] Suspending logging
2021-10-11 16:43:32 | stdout | 2021-10-11 18:43:32.887 car_id=1 [info] Fetching vehicle state ...
2021-10-11 16:45:04 | stdout | 2021-10-11 18:45:04.265 [error] Vehicle Error: Vehicle is offline
2021-10-11 16:45:34 | stdout | 2021-10-11 18:45:34.611 [error] Vehicle Error: Vehicle is offline
2021-10-11 16:46:05 | stdout | 2021-10-11 18:46:05.045 [error] Vehicle Error: Vehicle is offline
2021-10-11 16:46:35 | stdout | 2021-10-11 18:46:35.440 [error] Vehicle Error: Vehicle is offline
2021-10-11 16:47:05 | stdout | 2021-10-11 18:47:05.894 [error] Vehicle Error: Vehicle is offline
2021-10-11 16:47:36 | stdout | 2021-10-11 18:47:36.249 [error] Vehicle Error: Vehicle is offline
2021-10-11 16:48:06 | stdout | 2021-10-11 18:48:06.766 [error] Vehicle Error: Vehicle is offline
2021-10-11 16:48:36 | stdout | 2021-10-11 18:48:36.767 car_id=1 [info] Fetching vehicle state ...
2021-10-11 16:48:37 | stdout | 2021-10-11 18:48:37.116 [error] Vehicle Error: Vehicle is offline
2021-10-11 16:48:37 | stdout | 2021-10-11 18:48:37.121 car_id=1 [info] Start / :offline
2021-10-11 16:48:37 | stdout | 2021-10-11 18:48:37.339 car_id=1 [info] Disconnecting ...
2021-10-11 17:43:28 | stdout | 2021-10-11 19:43:28.845 car_id=1 [info] Start / :online


</body>
</html>

So there it is. I am looking for any ideas/input/shared experiences of what this could be. I know the MCU2 (and/or AP3 upgrade) definitely caused this because this issue never happened in the previous 3+ years of ownership with the exact same “parameters” and looking at my car's state graph is enough to tell the day the upgrade took place.

**Comment by BriCo84** [comment]
With Teslamate on, my car flips back online about every 75 minutes or so.

I confirmed Tesla, when everything else being equal, with Teslamate off, over a 4 day period, my car flipped online at very long intervals.  For ecxample
2 hour interval
then 3 hour interval
then it didn't flip online again they said until about 38 hours later
Then again 24 hours later

Something from Teslamate seems to only affect certain models and years.  I don't have this problem with my model y on the same setup, just my 2017 S75.  Everyone else with this issue also seems to have a 2017 S 75 or 75d.

**Reply by cleanev** [reply]
@BriCo84 - I added additional data point to support your claim and as always Tesla just washes their hands off stating I shouldn’t be using API’s continuously!

Here’s a link to your issues topic - https://github.com/adriankumpf/teslamate/issues/2556#issuecomment-1106758972
